crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1074,8 +1082,109 @@ pub(crate) async fn analyze_ecmascript_module(
             // the object allocation.
             let mut first_import_meta = true;
 
-            for effect in effects.into_iter() {
+            let mut queue = Vec::new();
+            queue.extend(effects.into_iter().rev());
+
+            while let Some(effect) = queue.pop() {
                 match effect {
+                    Effect::Conditional {
+                        condition,
+                        kind,
+                        ast_path: condition_ast_path,
+                        span: _,
+                    } => {
+                        let condition = link_value(condition).await?;
+                        macro_rules! inactive {
+                            ($block:ident) => {
+                                analysis.add_code_gen(UnreachableVc::new(AstPathVc::cell(
+                                    $block.ast_path.to_vec(),
+                                )));
+                            };
+                        }
+                        macro_rules! condition {
+                            ($expr:expr) => {
+                                analysis.add_code_gen(ConstantConditionVc::new(
+                                    Value::new($expr),
+                                    AstPathVc::cell(condition_ast_path.to_vec()),
+                                ));
+                            };
+                        }
+                        macro_rules! active {
+                            ($block:ident) => {
+                                queue.extend($block.effects.into_iter().rev());
+                            };
+                        }
+                        match *kind {
+                            ConditionalKind::If { then } => {
+                                if let Some(value) = condition.is_truthy() {
+                                    if value {
+                                        condition!(ConstantConditionValue::Truthy);
+                                        active!(then);
+                                    } else {
+                                        condition!(ConstantConditionValue::Falsy);
+                                        inactive!(then);
+                                    }
+                                } else {
+                                    active!(then);
+                                }
```

Comment:
These would be clearer as a match on the ternary:

```suggestion
                                match condition.is_truthy() {
                                    Some(true) => {
                                        condition!(ConstantConditionValue::Truthy);
                                        active!(then);
                                    } 
                                    Some(false) => {
                                        condition!(ConstantConditionValue::Falsy);
                                        inactive!(then);
                                    }
                                    None => {
                                        active!(then);
                                    }
                                }
```